src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -92,15 +92,27 @@ DatabaseShardImp::init()
                     seq))
             {
                 std::uint32_t seq2;
-                if (get_if_exists<std::uint32_t>(
-                        section,
-                        "earliest_seq", seq2) &&
-                    seq != seq2)
+                if (get_if_exists<std::uint32_t>(section, "earliest_seq", seq2))
                 {
-                    return fail("and [" + ConfigSection::nodeDatabase() +
-                        "] both define 'earliest_seq'");
+                    if (seq != seq2)
+                    {
+                        return fail("and [" + ConfigSection::nodeDatabase() +
+                            "] define different 'earliest_seq' values");
+                    }
+                }
+                else if (seq != XRP_LEDGER_EARLIEST_SEQ)
+                {
+                    return fail("doesn't define 'earliest_seq' while ["
+                        + ConfigSection::nodeDatabase() +"] does");
                 }
             }
+            else if (get_if_exists<std::uint32_t>(
+                    section, "earliest_seq", seq) &&
+                seq != XRP_LEDGER_EARLIEST_SEQ)
+            {
+                return fail("defines 'earliest_seq' while [" +
+                    ConfigSection::nodeDatabase() + "] doesn't");
+            }
```

Comment:
I keep forgetting that the field value is set in the `Database` constructor, and get stuck here thinking the value checked, but never used. Could you add a comment basically explaining / reminding readers of that?